### PR TITLE
Decoder is corrected, and both MUX ,Decoder, Encoder, Demux with tests added

### PR DESCRIPTION
**Pull request by rajat974**
Decoder was implemented as MUX, Now it is corrected. with tests and README changed.

### CODE REVIEW

**Comment by jayrambhia** [comment]
Heard about PEP8 and indentation?

**Comment by salilkapur** [comment]
Nice work!

Can you make these compatible with the Connector class?
ICs should be capable of redirecting their output to a Connector class object. See setOutput function in logic gates as a reference.

**Comment by rajat974** [comment]
Yes sure ! 
Started working on the same

**Comment by rajat974** [comment]
@salil93  please review the commits !

File: BinPy/Combinational/combinational.py [hunk 1]
```diff
@@ -1,24 +1,293 @@
-class Decoder:
+from BinPy import *
```
Comment (jayrambhia) [inline BinPy/Combinational/combinational.py]: is this necessary? to import everything?

File: BinPy/Combinational/combinational.py [hunk 2]
```diff
@@ -1,24 +1,293 @@
...
+import math
 
-	def __init__(self,inputs):
+class MUX(GATES):
+    """
+    This class can be used to create MUX in your circuit.
```
Comment (jayrambhia) [inline BinPy/Combinational/combinational.py]: Improve docs.
Add description.Input params. output params. some note if you think it's necessary. short example too.

**Comment by rajat974** [comment]
@jayrambhia  please have a look at it now !

**Comment by jayrambhia** [comment]
@rajat974 This some neat work. But I can't merge this directly. Make a new branch from this branch and rebase it to BinPy's develop branch and send another pull request. One more thing, we are trying to keep segregated tests. See this https://github.com/BinPy/BinPy/pull/86. So according to that put these tests in different files and add it in .travis.yml. And make sure that everything is passing.

**Pull request closed without merging**